android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -202,28 +203,66 @@ public Object getSelectionKey()
 				}
 			},
 			StorageStrategy.createLongStorage()
-		)
-			.withSelectionPredicate(SelectionPredicates.<Long>createSelectSingleAnything())
-			.build();
-		this.tracker.addObserver(new SelectionTracker.SelectionObserver()
-		{
+		);
 
-			@Override
-			public void onSelectionChanged()
-			{
-				super.onSelectionChanged();
+		if (proxy.getProperties().optBoolean(TiC.PROPERTY_EDITING, false)
+			&& proxy.getProperties().optBoolean(TiC.PROPERTY_ALLOWS_SELECTION_DURING_EDITING, false)) {
+			if (proxy.getProperties().optBoolean(TiC.PROPERTY_ALLOWS_MULTIPLE_SELECTION_DURING_EDITING, false)) {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectAnything())
+					.build();
+			} else {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectSingleAnything())
+					.build();
+			}
+
+			if (this.tracker != null) {
+				this.tracker.addObserver(new SelectionTracker.SelectionObserver()
+				{
+					int selectionCount = 0;
+
+					@Override
+					public void onSelectionChanged()
+					{
+						super.onSelectionChanged();
+
+						if (tracker.hasSelection() && selectionCount != tracker.getSelection().size()) {
```

Comment:
This if-condition's `selectionCount != tracker.getSelection().size()` part prevents single-selection mode from firing events correctly. Only the 1st single selection event fires, but subsequent events do not fire when selecting different rows since the selection count is always 1.

Also, iOS does not fire an "itemsselected" event when in single selection mode. iOS only fires the "itemclick" event.